api/src/main/java/org/pmiops/workbench/api/BigQueryService.java

```diff
@@ -0,0 +1,86 @@
+package org.pmiops.workbench.api;
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.FieldValue;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.QueryResponse;
+import com.google.cloud.bigquery.QueryResult;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Provider;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+@Service
+public class BigQueryService {
+
+    @Autowired
+    private BigQuery bigquery;
+
+    @Autowired
+    private Provider<WorkbenchConfig> workbenchConfig;
+
+    /**
+     * Execute the provided query using bigquery.
+     *
+     * @param query
+     * @return
+     */
+    public QueryResult executeQuery(QueryJobConfiguration query) {
+
+        // Execute the query
+        QueryResponse response = null;
+        try {
+            response = bigquery.query(query);
+        } catch (InterruptedException e) {
+            throw new BigQueryException(500, "Something went wrong with BigQuery: " + e.getMessage());
+        }
+
+        // Wait for the job to finish
+        while (!response.jobCompleted()) {
```

Comment:
So two problems with this:

1) It could run forever. (There's no terminating condition.)
2) It polls repeatedly instead of having BigQuery tell us when the results are in.

I think maybe we should do something like this:

https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/InsertDataAndQueryTable.java#L89

where we specify a timeout and throw an exception if the results aren't ready by the time query() returns.